### PR TITLE
Added ChatUI branding & put theming behind an env variable

### PR DESCRIPTION
**Pull request by nsarrazin**
We want to be able to offer ChatUI as its own thing decoupled from HuggingChat. This lets us do that easily.

We added an environment variable `PUBLIC_THEME` which will display the HuggingChat theme if set to `huggingchat` otherwise it will display the ChatUI theme.

* Use the ChatUI logo or the HuggingChat logo depending on env.
* Use the name "ChatUI" or "HuggingChat" depending on env.
* Added a `primary` color to tailwind config based on env theme. (yellow for HC, blue for ChatUI)
* Replaced all uses of `yellow` by the use of the `primary` color. 
* Added the right favicon & html page title based on env.
* Hide data sharing & about+privacy if not using HuggingChat theme

| Login  | Home  |
| ------------- | ------------- |
| image | image |

## Testing plan

1. No login, chat ui
leave `PUBLIC_THEME=` empty and comment out the `OPENID_*` keys

2. No login, huggingchat
`PUBLIC_THEME=huggingchat` and comment out the `OPENID_*` keys 

3. bring back login
Put back the `OPENID_*` keys and try with both `PUBLIC_THEME=` and `PUBLIC_THEME=huggingchat` 

If you want to get rid of announcement banners, just remove the var `PUBLIC_ANNOUNCEMENT_BANNERS` in .env.

### CODE REVIEW

**Comment by gary149** [comment]
I think instead of having exceptions everywhere ("isHuggingChat"), could we make things more configurable? Like having env variables for logo/main color/sharing/... Define the ChatUI ones in env and overwrite the local env for HuggingChat?

```
{#if isHuggingChat}
     <Logo classNames="text-3xl mr-1.5" />HuggingChat
{:else}
     <LogoChatUi classNames="text-3xl mr-1.5" />ChatUI
{/if}
```

this is better:

```
<Logo path={LOGO_PATH} classNames="text-3xl mr-1.5" />{APP_NAME}
```

**Comment by nsarrazin** [comment]
Yes it would be nicer if we only had the logo and the name but what about:
* the color (yellow vs blue)
* the favicons (we have to give a path)
* and other cases where we don't just add a logo or name, like for example in [the settings modal where we show the data sharing toggle only if it's HuggingChat ](https://github.com/huggingface/chat-ui/pull/298/commits/0ec0c68527db74cd795f7b0c6271df18619cb9d0#diff-60f5886632c25b3261f3a2def90144693b95de98b053d4e334399c2fae9c2951R36-R65)

Seems like it would be a lot of environment variable for each case, so for now I just handled it through  `isHuggingChat` but if you think there's a way to handle all the cases I don't mind changing it!

**Comment by gary149** [comment]
> the color (yellow vs blue)

It's even better, you'll be able to choose any tailwind color on Space duplication! (cool feature 🥳 )

> the favicons (we have to give a path)

Yes, you give a path and you can store HuggingChat and ChatUI assets in the repo btw.

> [the settings modal where we show the data sharing toggle only if it's HuggingChat](https://github.com/huggingface/chat-ui/pull/298/commits/0ec0c68527db74cd795f7b0c6271df18619cb9d0#diff-60f5886632c25b3261f3a2def90144693b95de98b053d4e334399c2fae9c2951R36-R65)

I don't see why it's not feasible you can start with a single DATA_SHARING variable set to false or even extend to multiple ones it if you want something more fine-grained.

**Comment by nsarrazin** [comment]
Alright no problem, I wasn't sure if we wanted to add all those vars, but if we're okay with it then it's indeed more configurable, gonna do it now!

**Comment by nsarrazin** [comment]
@gary149 should be good now! You can try it with

```
PUBLIC_APP_NAME=Chat UI
PUBLIC_APP_ASSETS=chatui
PUBLIC_APP_COLOR=blue
PUBLIC_APP_DATA_SHARING=
```

or 

```
PUBLIC_APP_NAME=HuggingChat
PUBLIC_APP_ASSETS=huggingchat
PUBLIC_APP_COLOR=yellow
PUBLIC_APP_DATA_SHARING=1
```

File: src/lib/components/LoginModal.svelte [hunk 1]
```diff
@@ -32,20 +35,27 @@
...
 		>
 			{#if $page.data.requiresLogin}
 				<button
 					type="submit"
-					class="mt-2 flex items-center whitespace-nowrap rounded-full bg-black px-5 py-2 text-lg font-semibold text-gray-100 transition-colors hover:bg-yellow-500"
+					class="mt-2 flex items-center whitespace-nowrap rounded-full bg-black px-5 py-2 text-lg font-semibold text-gray-100 transition-colors hover:bg-primary-500"
 				>
-					Sign in with <LogoHuggingFaceBorderless classNames="text-xl mr-1 ml-1.5" /> Hugging Face
+					Sign in
+					{#if PUBLIC_APP_ASSETS == "huggingchat"}
```
Comment (gary149) [inline src/lib/components/LoginModal.svelte:56]: I think I'd rather use `PUBLIC_APP_NAME` than `PUBLIC_APP_ASSETS` for this.

File: src/lib/components/NavMenu.svelte [hunk 2]
```diff
@@ -75,18 +75,20 @@
...
-	>
-		Feedback
-	</a>
-	<a
-		href="{base}/privacy"
-		class="flex h-9 flex-none items-center gap-1.5 rounded-lg pl-3 pr-2 text-gray-500 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-700"
-	>
-		About & Privacy
-	</a>
+	{#if PUBLIC_APP_ASSETS === "huggingchat"}
```
Comment (gary149) [inline src/lib/components/NavMenu.svelte:78]: Same maybe use `PUBLIC_APP_NAME`

File: src/lib/components/chat/ChatIntroduction.svelte [hunk 3]
```diff
@@ -31,8 +31,8 @@
 	<div class="lg:col-span-1">
 		<div>
 			<div class="mb-3 flex items-center text-2xl font-semibold">
-				<Logo classNames="mr-1 text-yellow-400 text-4xl flex-none" />
-				HuggingChat
+				<Logo classNames="mr-1 w-[30px] h-[30px]" />
```
Comment (gary149) [inline src/lib/components/chat/ChatIntroduction.svelte:34]: Same for width/height

File: tailwind.config.cjs [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
+dotenv.config({ path: "./.env" });
 
 /** @type {import('tailwindcss').Config} */
 export default {
 	darkMode: "class",
 	content: ["./src/**/*.{html,js,svelte,ts}"],
 	theme: {
 		extend: {
+			colors: {
+				primary: colors[process.env.PUBLIC_APP_COLOR],
```
Comment (gary149) [inline tailwind.config.cjs:14]: 👍

**Review by gary149 — CHANGES_REQUESTED** [review]
Made some comments but yes that's better.

File: src/lib/components/NavMenu.svelte [hunk 5]
```diff
@@ -23,8 +23,8 @@
 
 <div class="sticky top-0 flex flex-none items-center justify-between px-3 py-3.5 max-sm:pt-0">
 	<a class="flex items-center rounded-xl text-lg font-semibold" href="{PUBLIC_ORIGIN}{base}/">
-		<Logo classNames="mr-1 text-3xl" />
-		HuggingChat
+		<Logo classNames="mr-1 w-[30px] h-[30px]" />
```
[the remark this answers is not in the document]
Comment (nsarrazin) [reply src/lib/components/NavMenu.svelte]: the favicon we use for chatui & huggingchat had different width & height, but I made them equal now :+1:

File: src/lib/components/LoginModal.svelte [hunk 6]
```diff
@@ -2,23 +2,26 @@
...
 </script>
 
 <Modal>
 	<div
-		class="flex w-full flex-col items-center gap-6 bg-gradient-to-t from-yellow-500/40 via-yellow-500/10 to-yellow-500/0 px-4 pb-10 pt-9 text-center"
+		class="flex w-full flex-col items-center gap-6 bg-gradient-to-t from-primary-500/40 via-primary-500/10 to-primary-500/0 px-4 pb-10 pt-9 text-center "
 	>
 		<h2 class="flex items-center text-2xl font-semibold text-gray-800">
-			<Logo classNames="text-3xl mr-1.5" />HuggingChat
+			{PUBLIC_APP_NAME}
```
[the remark this answers is not in the document]
Comment (nsarrazin) [reply src/lib/components/LoginModal.svelte:21]: fixed

File: src/lib/components/LoginModal.svelte (hunk 1 above)
Comment (nsarrazin) [reply src/lib/components/LoginModal.svelte]: ok, done!

File: src/lib/components/chat/ChatIntroduction.svelte (hunk 3 above)
Comment (nsarrazin) [reply src/lib/components/chat/ChatIntroduction.svelte]: see above, fixed!

File: src/lib/components/chat/ChatIntroduction.svelte [hunk 7]
```diff
@@ -31,7 +31,7 @@
 	<div class="lg:col-span-1">
 		<div>
 			<div class="mb-3 flex items-center text-2xl font-semibold">
-				<Logo classNames="mr-1" />
+				<Logo classNames="mr-1 min-w-fit" />
```
Comment (gary149) [inline src/lib/components/chat/ChatIntroduction.svelte:34]:
```suggestion
				<Logo classNames="mr-1 flex-none" />
```

**Review by gary149 — APPROVED** [review]

**Pull request merged**